### PR TITLE
Use Sawyer

### PR DESCRIPTION
**Pull request by pezholio**
This tweaks the gem to use [Sawyer](https://github.com/lostisland/sawyer) to access the API, rather than just Faraday. This gives us some nice sugar on top, as well as makes it easier to do things like pagination (which this PR also handles)

### CODE REVIEW

File: breathe.gemspec [hunk 1]
```diff
@@ -34,5 +34,5 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "pry", "~> 0.12.2"
   spec.add_development_dependency "webmock", "~> 3.7.6"
 
-  spec.add_dependency "faraday", "~> 0.17.0"
+  spec.add_dependency "sawyer", "~> 0.8.2"
```
Comment (tahb) [inline breathe.gemspec]: I don't know anything about Sawyer to be honest, but what made you want to switch and what problem does this solve?

File: lib/breathe/client.rb [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
-      @_absences ||= Absences.new(self)
-    end
-
-    def get(url, url_opts = {})
-      request(:get, url, url_opts)
+      @absences ||= Absences.new(self)
     end
 
-    private
+    def response(type, path, args)
```
Comment (tahb) [inline lib/breathe/client.rb]: I do think named parameters are worth doing to protect against problems with parameter ordering and to make it a little clearer to the reader, removing the need to go and read the method to understand what each param means.

For example at the extreme of unhelpfulness we could have `foo("1", "", true)`. This is made much more manageable (IMO) when read as `foo(age: "1", name: "", teeth: true)`.. "Okay I need to provide a name! Duh"

File: lib/breathe/client.rb [hunk 3]
```diff
@@ -1,43 +1,45 @@
...
+      response = request(type, path, query: args)
 
-    attr_reader :api_key, :environment
+      return Response.new(response, path) if /^2+/.match?(response.status.to_s)
```
Comment (tahb) [inline lib/breathe/client.rb]: Could we query the response for a successful status code in a clearer way that might encourage a unit test or 2? Perhaps after we initialise our response object?

```
breathe_response = request(…)
response = Response.new(response: breathe_response, path: path)
return response if response.success?
```

File: lib/breathe/response.rb [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
   class Response
     extend Forwardable
-    attr_reader :data, :type, :response
+    attr_reader :response, :type
 
     delegate [:each, :find, :select, :count, :[]] => :body
 
     def initialize(response, type)
       @response = response
-      @data = JSON.parse(response.body)
```
Comment (tahb) [inline lib/breathe/response.rb]: Oo la la, so I guess Sawyer does this for us then and that's a help.

**Comment by pezholio** [comment]
Awesome, ta. Always nice to get a second pair of eyes on these things 👍 I've implemented a couple of tweaks now

**Pull request merged**